src/Microsoft.Framework.Logging.Debug/DebugLogger.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.Framework.Logging.Debug
+{
+    public class DebugLogger : ILogger
+    {
+        private readonly Func<string, LogLevel, bool> _filter;
+        private readonly string _name;
+
+        public DebugLogger(string name)
+            : this(name, filter: null)
+        {
+        }
+
+        public DebugLogger(string name, Func<string, LogLevel, bool> filter)
+        {
+            _name = string.IsNullOrEmpty(name) ? nameof(DebugLogger) : name;
+            _filter = filter ?? ((category, logLevel) => true);
+        }
+
+        public IDisposable BeginScopeImpl(object state)
+        {
+            return null;
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            if (!Debugger.IsAttached)
+            {
+                return false;
+            }
+
+            return _filter(_name, logLevel);
+        }
+
+        public void Log(LogLevel logLevel, int eventId, object state, Exception exception, Func<object, Exception, string> formatter)
+        {
+            if (!IsEnabled(logLevel))
+            {
+                return;
+            }
+
+            string message;
+            var values = state as ILogValues;
+            if (formatter != null)
+            {
+                message = formatter(state, exception);
+            }
+            else if (values != null)
+            {
+                var formattedMessage = new StringBuilder();
+                FormatLogValues(formattedMessage, values);
+                message = formattedMessage.ToString();
```

Comment:
curious, was it intentional that you excluded writing out `exception` because in debug mode one could break at exceptions?...like there are scenarios where exceptions are swallowed but a message with exception is still logged...so probably we want to include exception too?